slashCommands/vote.ts

```diff
@@ -12,187 +21,149 @@ export default {
 
-        let reactionAr = new Array(0);
-
         // --- Variables with input
         const options = interaction.options as CommandInteractionOptionResolver;
-
-        let time = options.getInteger("votetime", false);
-
-        let timed = options.getBoolean("istimed");
-        timed = (timed == null) ? false : timed;
-
-        let title = options.getString("name");
-
-        // --- Check for undefined
-
-        time = ((time == undefined || time <= 0) ? 30 : time);
-
-        title = (title == undefined) ? "" : title;
-
-        // --- group variables
+        const time = options.getInteger("votetime", false);
+        const timed = time != undefined;
+        const title = options.getString("name", true);
 
         const guid = get("vote_group", "config") as string;
         const maingroup = interaction.guild.roles.cache.get(guid);
 
         // --- create initial system for time or group voting
 
-        if (timed == true) {
-            msg += "This is a timed vote. The vote is running " + time + " Seconds!";
-            maxLoop = 1;
+        if (timed) {
+            msg += "This is a timed vote. The vote is running " + time + " Seconds!\n";
+            msg += "This vote ends <t:" + Math.ceil(Date.now()/1000 + time) + ":R> \n"
         } else {
             // get online member
             const groupmembers = maingroup.members.map(m=>m.user.id);
-            usedVotes = Math.round(groupmembers.length/2);
+            usedVotes = Math.ceil(groupmembers.length/2);
             // further variables set
             usedVotes = ((usedVotes == 0) ? 1 : usedVotes);
             msg += "This is a majority voting. " + usedVotes + " Votes required!";
         }
 
-        // --- MessageEmbed
-
-        const voteAN = new EmbedBuilder()
-            .setTitle((title == "") ? "Simple Voting ": title)
-            .setDescription(msg)
-            .setColor("#477ce0")
-            .addFields({name: "Maximal Runtime", value: "" + maxLoop * time + " Seconds", inline: true});
-
-        const reply = await interaction.reply({embeds: [voteAN], fetchReply:true}) as Message;
-
-
-        await reply.react(upEmo);       //react with emote up
-        await reply.react(downEmo);     //react with emote down
+        const reply = await interaction.reply(getEmbedOptions(title, msg)) as Message;
 
         // eslint-disable-next-line
-        const filter = (reaction, user) => { //filter for only getting the up and down emoji
-            return user.id != reply.author.id;
+        const filter = (interaction) => {
+            if (!interaction.isButton()) {
+                return false;
+            }
+            return interaction.message.id === reply.id;
         };
 
         //reaction controller
-        const collector = reply.createReactionCollector({filter, dispose: true} as ReactionCollectorOptions);
+        const collector = reply.createMessageComponentCollector({filter});
+        console.log(typeof collector);
```

Comment:
Floating console.log that should be removed

Reply:
removed in next commit